src/auth/src/credentials.rs

```diff
@@ -657,6 +616,28 @@ mod test {
 
     type TestResult = std::result::Result<(), Box<dyn std::error::Error>>;
 
+    pub(crate) fn get_token_from_headers(headers: &HeaderMap) -> String {
+        let token = headers.get(AUTHORIZATION).unwrap();
+        token
+            .to_str()
+            .unwrap()
+            .split_whitespace()
+            .nth(1)
+            .unwrap()
+            .to_string()
+    }
```

Comment:
I know this is test code, and crashes / assertions are fine, but if you return `Option<String>` you can simplify this code a lot:

```suggestion
    pub(crate) fn get_token_from_headers(headers: &HeaderMap) -> Option<String> {
        let token = headers.get(AUTHORIZATION)?;
        token
            .to_str()?
            .split_whitespace()
            .nth(1)
            .map(|v| v.to_string())
    }
```

And then the call sites can do an `.unwrap()` if that is appropriate.

Reply:
I think to_str() return back a result, so this won't work, but thanks for the suggestion, simplifying it using something like this

```rust
        headers.get(AUTHORIZATION)
            .and_then(|token_value| token_value.to_str().ok())
            .and_then(|s| s.split_whitespace().nth(1))
            .map(|s| s.to_string())
```